numpy/lib/npyio.py

```diff
@@ -85,8 +85,8 @@ def __init__(self, obj):
     def __getattribute__(self, key):
         try:
             return object.__getattribute__(self, '_obj')[key]
-        except KeyError:
-            raise AttributeError(key)
+        except KeyError as e:
+            raise AttributeError(key) from e
```

Comment:
```suggestion
        except KeyError:
            raise AttributeError(key) from None
```
Since the AttributeError already contains all the appropriate information

Reply:
@eric-wieser I committed your suggestion.